pallets/public-credentials/src/tests.rs

```diff
@@ -297,81 +516,128 @@ fn add_ctype_not_found() {
+	let credential_id: CredentialIdOf<Test> = CredentialIdOf::<Test>::default();
+	let deposit: Balance = <Test as Config>::Deposit::get();
 
 	ExtBuilder::default()
-		.with_balances(vec![(ACCOUNT_00, public_credential_deposit + attestation_deposit)])
-		.with_ctypes(vec![(attestation.ctype_hash, attester.clone())])
-		.with_attestations(vec![(claim_hash, attestation.clone())])
-		.with_public_credentials(vec![(subject_id, claim_hash, new_credential)])
+		.with_balances(vec![(ACCOUNT_00, deposit)])
+		.with_public_credentials(vec![(subject_id, credential_id, new_credential)])
 		.build()
 		.execute_with(|| {
 			assert_ok!(PublicCredentials::remove(
 				DoubleOrigin(ACCOUNT_00, attester.clone()).into(),
-				claim_hash,
-				None
+				credential_id,
+				None,
 			));
-			// Test interactions with attestation pallet
-			assert_eq!(Attestations::<Test>::get(&claim_hash), None);
 
 			// Test this pallet logic
-			assert_eq!(Credentials::<Test>::get(&subject_id, &claim_hash), None);
-			assert_eq!(CredentialsUnicityIndex::<Test>::get(&claim_hash), None);
+			assert_eq!(Credentials::<Test>::get(&subject_id, &credential_id), None);
+			assert_eq!(CredentialSubjects::<Test>::get(&credential_id), None);
 
 			// Check deposit release logic
 			assert!(Balances::reserved_balance(ACCOUNT_00).is_zero());
 
 			// Removing the same credential again will fail
 			assert_noop!(
-				PublicCredentials::remove(DoubleOrigin(ACCOUNT_00, attester.clone()).into(), claim_hash, None),
+				PublicCredentials::remove(DoubleOrigin(ACCOUNT_00, attester.clone()).into(), credential_id, None,),
 				Error::<Test>::CredentialNotFound
 			);
+		});
+}
+
+#[test]
+fn remove_same_attester_wrong_ac() {
+	let attester = sr25519_did_from_seed(&ALICE_SEED);
+	let wrong_submitter = sr25519_did_from_seed(&BOB_SEED);
+	let subject_id: <Test as Config>::SubjectId = SUBJECT_ID_00;
+	let mut new_credential = generate_base_credential_entry::<Test>(ACCOUNT_00, 0, attester.clone(), None);
+	new_credential.authorization_id = Some(attester.clone());
+	let credential_id: CredentialIdOf<Test> = CredentialIdOf::<Test>::default();
+	let deposit: Balance = <Test as Config>::Deposit::get();
 
-			// Adding only the attestation without the credential will also fail to remove
-			// the credential.
-			Attestation::add(
+	ExtBuilder::default()
+		.with_balances(vec![(ACCOUNT_00, deposit)])
+		.with_public_credentials(vec![(subject_id, credential_id, new_credential)])
+		.build()
+		.execute_with(|| {
+			assert_ok!(PublicCredentials::remove(
 				DoubleOrigin(ACCOUNT_00, attester.clone()).into(),
-				claim_hash,
-				attestation.ctype_hash,
-				None,
-			)
-			.expect("Adding the same attestation again should not fail");
+				credential_id,
+				Some(MockAccessControl(wrong_submitter))
+			));
+
+			// Test this pallet logic
+			assert_eq!(Credentials::<Test>::get(&subject_id, &credential_id), None);
+			assert_eq!(CredentialSubjects::<Test>::get(&credential_id), None);
```

Comment:
Better to use `is.none()`

```suggestion
			assert!(Credentials::<Test>::get(&subject_id, &credential_id),.is_none());
			assert!(CredentialSubjects::<Test>::get(&credential_id).is_none());
```

Reply:
Nice one! https://github.com/KILTprotocol/kilt-node/pull/392/commits/e95d50e7eee6f3542cd090e82930a842d5c0f0fd